.github/workflows/ci.yml

```diff
@@ -0,0 +1,66 @@
+name: CI
+
+on:
+  push:
+  pull_request:
+
+env:
+  CONSUL_VERSION: 1.6.1
+
+jobs:
+  build:
+    strategy:
+      matrix:
+        os: [ubuntu-latest, windows-latest, macos-latest]
+      fail-fast: false
+    runs-on: ${{ matrix.os }}
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+      - name: Setup .NET Core SDK
+        uses: actions/setup-dotnet@v1
+        with:
+          dotnet-version: 3.1.300
+      - name: Download Consul
+        shell: bash
+        run: |
+          cd Consul.Test
+          SYSTEM=$(uname | sed 's/MINGW.*/windows/' | tr A-Z a-z)
+          curl -sSL https://releases.hashicorp.com/consul/${CONSUL_VERSION}/consul_${CONSUL_VERSION}_${SYSTEM}_amd64.zip -o consul.zip
+          unzip consul.zip
+          rm consul.zip
+      - name: Build
+        run: dotnet build --configuration=Release -p:TreatWarningsAsErrors=true
+      - name: Repack
+        if: runner.os == 'Windows'
```

Comment:
Could use a comment explaining why you're doing this.

Reply:
Good point, I added comments to the CI build job.
FYI this is preserving existing behaviour of the previous CI.